package.json

```diff
@@ -16,7 +16,7 @@
     "build:ci": "yarn run build --mode preview -c ./vite.config.development.ts",
     "build:gui": "mkdir gui && mv dist/* gui && mv gui dist/gui && mv dist/gui/mockServiceWorker.js dist/",
     "build:preview": "yarn run build:ci && yarn build:gui && cp _redirects dist/_redirects",
-    "build:staging": "yarn build && yarn build:gui",
+    "build:staging": "VITE_ZONE_CREATION_FLOW=enabled yarn build && yarn build:gui",
```

Comment:
**Comment**: Not sure if there is a better place to put this.

Reply:
I rather avoid inverting this flag to "on by default" because it’s riskier. Forgetting to turn it off could leak this into a production build which can’t happen by accident if it’s "off by default".

Not sure what the alternative to setting it in the "build:staging" script is. Setting `KUMA_ZONE_CREATION_FLOW: 'enabled',` in `src/services/e2e.ts` doesn’t seem to work which I guess is owed to the fact that `build:staging` runs the production build using production.ts.

And I really do want to test this flow here because it’s where the code is (and where it can be broken).

Ideally, we could turn this flag on on a per-test basis so we could test both sides: enabled and disabled.

Reply:
Ideally the code would live in the place where we actually need it now (we could always just move it here when we want to enable it here). That would solve all of this.

I think another solution would be to add yet another run of tests on a dev build (i.e. `yarn dev`), which whilst isn't too much hassle to do, longer term it means we are complicating matters a little. 🤔 Actually, we could probably use a tag to only run the one set of tests in dev and exclude that same one set of tests from prod. In fact I think this is my favourite option considering everything/everyone. What do you think of adding a cucumber `@dev` tag (or similarly named) to your tests here, and then only running `@dev` tagged tests against `yarn dev` and anything else runs against `yarn build:staging`. The idea being, once this 'goes live' we just remove the `@dev` tag from the tests. We could even add a test testing the feature is not enabled (without the `@dev` tag of course)

Another option and perhaps my least favourite but easiest, is to just accept that we aren't really testing a production build, which is what we've done with this change here.

Just to be clear, my fave is the second option. What do you think of using a cucumber `@dev` tag?

Reply:
> This is to degree less convenient that I’m not willing to consider it (again).

😆  I'd guessed that, hence the second option being my fave.

> It seems like quite some work to add the logic for running against two different builds in CI.

Only running certain tests based on a tag can be done via env vars and the functionality is built into cucumber already. Pointing cypress to a different build is also an env var.

> Another option and perhaps my least favourite but easiest, is to just accept that we aren't really testing a production build, which is what we've done with this change here.

Sounds like you're not keen, so I think we should just go with the third option for the moment then. i.e. no action here if you are happy with this, thanks for the convo.

Oh I was going to ask it looks like we don't actually need `cross-env` is that right?

Reply:
I just don’t know how to add this tag-based dev build logic. We’d need to add support for a `@dev` tag (I assume the functionality being built into Cucumber refers to the ability to add custom tags not the presence of a `@dev` tag, yes?) and create two builds in two separate directories (I assume pointing Cypress to a different build refers to setting a different `KUMA_GUI_DIST_ROOT` for a dev CI build, yes?).

> Oh I was going to ask it looks like we don't actually need cross-env is that right?

_We_ technically don’t because our operating systems understand this syntax, but Windows doesn’t. cross-env abstracts this away so that one can run these commands on Windows systems. (Though that is currently broken through the use of `&&` which Windows also doesn’t understand.)

Reply:
> (I assume the functionality being built into Cucumber refers to the ability to add custom tags not the presence of a @dev tag, yes?)

yup

> We’d need to add support for a @dev tag

We just add `@dev` to the top of the test and then from what I remember we can specify which tags to run via env vars. Some deets here https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/tags.md

> (I assume pointing Cypress to a different build refers to setting a different KUMA_GUI_DIST_ROOT for a dev CI build, yes?).

Kinda, yup, I think its `KUMA_BASE_URL` (look in in `visit.ts`) i.e. we would run `yarn dev` and point Cypress to it using `KUMA_BASE_URL` (FYI: I made MSW auto disable whenever you point Cypress at it)

All in all this was a good chat, but I'm happy for us to leave as is for now if its not straight forward to do the `@dev` thing. I do think it would be a good "utility" for us to have in our testing toolkit though, so maybe something for the future.

> We technically don’t because our operating systems understand this syntax, but Windows doesn’t. cross-env abstracts this away so that one can run these commands on Windows systems. (Though that is currently broken through the use of && which Windows also doesn’t understand.)

I'd be happy to remove cross-env as a dependency if you also were, sounds like its just cruft. Maybe we make an issue for the removal?

Just to be clear especially considering all the conversation on this one, there's no request for change here unless you wanted to have a go at the `@dev` thing (even then we could do that in a PR at some point)

Reply:
I’d like to implement this, but I don’t have the time, so I’m leaving this as it is.